### ISSUE TITLE
After pytorch fixes the random seed, why can the results be repeated only in the first few epochs ?

### ISSUE THREAD

**Issue by pancccool** [pytorch/pytorch#70537]
## Issue description
After pytorch fixes the random seed, only the results of the first few epochs can be repeated, and the results of the next few epochs cannot be repeated.

## Code example
Here is my seed setup:
jj=0
np.random.seed(jj)
random.seed(jj)
torch.manual_seed(jj)
torch.cuda.manual_seed_all(jj)
torch.cuda.manual_seed(jj)
torch.backends.cudnn.benchmark = False
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.enabled = False
os.environ[‘PYTHONHASHSEED’] = str(jj)
os.environ[‘CUBLAS_WORKSPACE_CONFIG’] = ‘:4096:8’
Furthermore，the Dataloader:
def seed_worker(worker_id):
      worker_seed = torch.initial_seed() % 2 ** 32
      np.random.seed(worker_seed)
      random.seed(worker_seed)
data_loader= DataLoader(train_sample, batch_size=20,num_workers=0,worker_init_fn=seed_worker)

When I set epochs=10, the results of the experiment are the same no matter how many times it is repeated. When I set epochs=200, the experimental results have no deterministic characteristics.

- OS:
- PyTorch version: 1.8.1
- Python version:. 3.6
- CUDA/cuDNN version: 11.1
- GPU models and configuration: RTX3070

cc @SsnL @VitalyFedyunin @ejguan @NivekT @mruberry @kurtamohler

**Comment by VitalyFedyunin** [comment]
Hi! Do you have minimal reproducible example?

**Comment by SsnL** [comment]
be aware of https://pytorch.org/docs/stable/notes/randomness.html#avoiding-nondeterministic-algorithms

**Comment by kurtamohler** [comment]
Another note, since I see that you're setting `CUBLAS_WORKSPACE_CONFIG` with `os.environ`. That usually doesn't work. The workspace is configured once, during `import torch`. So you'll either have to move `
os.environ[‘CUBLAS_WORKSPACE_CONFIG’] = ‘:4096:8’` above `import torch` or set `CUBLAS_WORKSPACE_CONFIG` in your environment before running python

**Comment by kurtamohler** [comment]
I believe the issue has been addressed. We can reopen if not
